internal/clients/http/ec2/ec2_client_test.go

```diff
@@ -4,6 +4,61 @@ import (
 	"testing"
 )
 
+var expected = Statement{
+	Effect: "Allow",
+	Action: []string{
+		"iam:GetPolicyVersion",
+		"iam:GetPolicy",
+		"ec2:CreateKeyPair",
+		"ec2:CreateLaunchTemplate",
+	},
+}
+
+var missingLastPermission = []Statement{
+	{
+		Effect: "Allow",
+		Action: []string{
+			"iam:GetPolicyVersion",
+			"iam:GetPolicy",
+		},
+	},
+	{
+		Effect: "Allow",
+		Action: []string{
+			"ec2:CreateKeyPair",
+		},
+	},
+}
+
+var duplicityPermissions = []Statement{
+	{
+		Effect: "Allow",
+		Action: []string{
+			"iam:GetPolicyVersion",
+			"iam:GetPolicy",
+		},
+	},
+	expected,
+}
+
+func TestListMissingPermissions(t *testing.T) {
+	missingStatements := listMissingPermissions(missingLastPermission, expected)
+	if len(missingStatements) != 1 || missingStatements[0] != expected.Action[len(expected.Action)-1] {
+		t.Errorf(`missing permission listing failed`)
+	}
+
+	expectedStatementList := []Statement{expected}
+	shouldBeEmpty := listMissingPermissions(expectedStatementList, expected)
+	if len(shouldBeEmpty) != 0 {
+		t.Errorf(`correct missing permissions listing failed`)
+	}
+
+	shouldBeEmpty = listMissingPermissions(duplicityPermissions, expected)
+	if len(shouldBeEmpty) != 0 {
+		t.Errorf(`missing permission listing with duplicities failed`)
+	}
+}
```

Comment:
I suggest to use `require` package from the `testify` library, tests are much shorter.